frontend/lib/src/WidgetStateManager.ts

```diff
@@ -709,6 +713,32 @@ export class WidgetStateManager {
       this.props.formsDataChanged(this.formsData)
     }
   }
+
+  public setExtraWidgetInfo(
+    widget: WidgetInfo,
+    key: string,
+    value: any
+  ): void {
+    if (!this.extraWidgetInfo.has(widget.id)) {
+      this.extraWidgetInfo.set(widget.id, new Map<string, any>())
+    }
+
+    const currentMap = this.extraWidgetInfo.get(widget.id)
+
+    if (currentMap) {
+      currentMap.set(key, value)
+    }
```

Comment:
Wouldn't the `if` statement below be redundant? We know for sure that `currentMap` must be set because we would have just done so above. It'd probably be preferable to cast the return value of `this.extraWidgetInfo.get(widget.id)` if typescript isn't smart enough to realize this